### PR TITLE
[WIP] Scripts/TrialOfTheChampion: Full rework

### LINKED ISSUE #383: [Trial of Champion] Event bugged.
-Start the event, first encounter shows up , but they won't change their flags, they just stay there unatackable.If i change their flags , they will start and attack (low damage per hit).
Moving to second phase of the first encounter , the three bosses act normal , you kill them but the chest doesn't spawn(not even in other phases).

-After this the event starter gets his npcflag set to 0. I will change his npcflag to 1 , i start the next encounter , kill the boss ,same problem the chest isn't there.

-At the final boss , the event starter acts the same , resets his npcflag to 0.Set npcflag to 1 start the event , The Black Knight acts good scripted , has the loot set.

### PR DESCRIPTION
**Pull request by Aokromes**
Since xerkoss stopped to work on this here goes one updated version.
Some coding styles, remove incorrect (C) headers, use proper mobs, fix sql and some code.
Original work by xerkoss, updated by dduarte, Killyana and me.

TODO:
Make core to remove flags of caches on boss kill.
Make core change faction of few mobs on runtime.
Starting instance with horde shows alliance mobs on horde mounts.
You can attack mobs on any group before the timed event.
the 3 groups moves to instance big door instead to center.
After killing the 3 groups of small mobs champions are "killed" and mounts attacks you being unkillable.

Updated version of #14478 closes #383

### CODE REVIEW

**Comment by Aokromes** [comment]
@Mack36  test this :P

File: src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp [hunk 1]
```diff
@@ -135,9 +143,78 @@ class instance_trial_of_the_champion : public InstanceMapScript
...
+                    {
+                        champion1->AI()->DoAction(1);
+                        champion2->AI()->DoAction(1);
+                        champion3->AI()->DoAction(1);
+                        
+                        for (GuidList::const_iterator itr = VehicleList.begin(); itr != VehicleList.end(); ++itr)
+                            if (Creature* vehicles = instance->GetCreature(*itr))
+                                vehicles->DespawnOrUnsummon();
+                    }
+                    DoRemoveAurasDueToSpellOnPlayers(66482); // Defender
```
Comment (MitchesD) [inline src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/instance_trial_of_the_champion.cpp]: don't use magic number

**Comment by Mack36** [comment]
Problems I noticed:
1. VS reported linking error in `boss_grand_champions.cpp` because of
```` c++
virtual void Initialize() = 0;
````
Changed to `virtual void Initialize() {}` to compile.
2. Champions' mounts have `NOT_SELECTABLE` flag, e.g. npc 35641 Zul'tore's Mount has `unit_flag = 33587456`,
which equals to `IMMUNE_TO_PC, UNK_15 and NOT_SELECTABLE` flags. Should be removed somewhere after summoning.
3. There's a difference between normal and heroic versions of champion bosses, e.g. 34705 (heroic version 36088) Marshal Jacob Alerius (Grand Champion of Stormwind) has `unit_flag = 33554690 (NON_ATTACKABLE, IMMUNE_TO_PC, NOT_SELECTABLE)`, so on normal mode he's not selectable until trash mobs are killed. But on a heroic mode he has `unit_flag = 0`, so he can even be killed before trash mobs. 
I don't know, probably that was intended, but this seems to be a wider problem, because in any other instance unit flags. immunity masks and others are not being loaded from the basic (normal) creature entry, 
4. If players stay far from the center of the room while Black Knight is landing, he will become non killable (immune to pc), and event will just stop at this point.
5. There are no speech sounds of Announcer, Varian, Tirion, Thrall and others, but their speeches are still written in chat (sounds are enabled, probably caused by some local problem, or maybe by ruRU localisation?).

**Comment by Frytiks** [comment]
I don't have speech too.

**Comment by Aokromes** [comment]
https://github.com/TrinityCore/TrinityCore/pull/14893#issuecomment-114088164 
1) yeah, on 3.3.5a that row must be changed, this source was made originally on 6.x branch and because that i followed the work there.
2, 3,5) it looks db must be updated for those mobs.

**Comment by Aokromes** [comment]
#14938

**Pull request closed without merging**